### PR TITLE
Fix navigation menu items reordering issue

### PR DESCRIPTION
**Pull request by kamilpastuszka**
I want to merge this change so that navigation menu can keep its state after reordering.

 There was a need to upgrade fast-array-diff module to the latest version as one of the functions returned incorrect values.

**PR intended to be tested with API branch:** 

### Pull Request Checklist

1. [ ] This code contains UI changes
2. [ ] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `[data-test-id]` are added for new elements
4. [ ] Changes are mentioned in the changelog
5. [ ] The changes are tested in different browsers and in light/dark mode

### Test environment config

API_URI=https://qa.staging.saleor.cloud/graphql/

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-2647 MenuItems reordering nor working](https://app.clickup.com/t/2549495/SALEOR-2647)

**Review by krzysztofwolski — CHANGES_REQUESTED** [review]
No issues with the code, but testing on test environment still has issues.

https://saleor-2647-menuitems-reordering-nor-working.dashboard.saleor.rocks/navigation/TWVudToxMjg%3D
Base menu setup:
image

Now change order of two submenus and click save:
image

After refresh changes are dissapearing. Now I cannot move AA to the first supposition (to achieve AA, AB, AC

File: src/navigation/components/MenuDetailsPage/MenuDetailsPage.tsx [hunk 1]
```diff
@@ -69,7 +69,7 @@ const MenuDetailsPage: React.FC<MenuDetailsPageProps> = ({
     });
 
     if (result) {
-      setTreeOperations([]);
+      setTreeOperations([...treeOperations]);
```
Comment (jwm0) [inline src/navigation/components/MenuDetailsPage/MenuDetailsPage.tsx:72]: I don't see how that's doing anything, you're setting current state as new state.

**Review by jwm0 — CHANGES_REQUESTED** [review]

File: src/navigation/components/MenuDetailsPage/MenuDetailsPage.tsx (hunk 1 above)
Comment (kamilpastuszka) [reply src/navigation/components/MenuDetailsPage/MenuDetailsPage.tsx:72]: With empty array, as it was before, navigation items would move back to their previous location. 
I realized that the if statement block there is redundant - I will remove it next time.

**Comment by kamilpastuszka** [comment]
It will be looked into again after an issue is fixed on the back-end.

**Pull request closed without merging**